### ISSUE TITLE
PostgreSQL won't start: undefined symbol: arc4random

### ISSUE THREAD

**Issue by gtod** [tanglebones/pg_tuid#4]
Hey Cliff,

Thanks for your work on this library.  @jesseschalken who is working with me on a project wrote [this](https://gist.github.com/jesseschalken/d1db6b0712c79a0b125122945b405ca1) up yesterday as we tried to understand the importance of the work you have done.  And we decided it was important. ;)

Today I downloaded the source to my Debian box and built and installed it but Postgres would not restart:
```
2018-02-22 10:25:06 AEDT [12138-1] FATAL:
  could not load library "/usr/lib/postgresql/10/lib/tuid.so":
 /usr/lib/postgresql/10/lib/tuid.so: undefined symbol: arc4random
```
I spent an hour or so trying to work out just whose responsibility it is to ensure that Postgres (and your extension) have access to `libbsd.so.0` but I've been too long away from C development to get a quick answer.

Do you have any suggestions please?

My postgresql package is is prebuilt for debian from http://apt.postgresql.org/pub/repos/apt/
```
castor ~/postgres/pg_tuid/pg_c/% pg_config --version  
PostgreSQL 10.2 (Debian 10.2-1.pgdg90+1)

castor ~/postgres/pg_tuid/pg_c/% pg_config --libs   
-lpgcommon -lpgport -lpthread -lselinux -lxslt -lxml2 -lpam -lssl -lcrypto -lgssapi_krb5 -lz -ledit -lrt -lcrypt -ldl -lm  

castor ~/postgres/pg_tuid/pg_c/% pg_config --configure
'--with-icu' '--with-tcl' '--with-perl' '--with-python' '--with-pam' '--with-openssl' '--with-libxml' '--with-libxslt' '--with-tclconfig=/usr/lib/x86_64-linux-gnu/tcl8.6' '--with-includes=/usr/include/tcl8.6' 'PYTHON=/usr/bin/python' '--mandir=/usr/share/postgresql/10/man' '--docdir=/usr/share/doc/postgresql-doc-10' '--sysconfdir=/etc/postgresql-common' '--datarootdir=/usr/share/' '--datadir=/usr/share/postgresql/10' '--bindir=/usr/lib/postgresql/10/bin' '--libdir=/usr/lib/x86_64-linux-gnu/' '--libexecdir=/usr/lib/postgresql/' '--includedir=/usr/include/postgresql/' '--with-extra-version= (Debian 10.2-1.pgdg90+1)' '--enable-nls' '--enable-integer-datetimes' '--enable-thread-safety' '--enable-tap-tests' '--enable-debug' '--disable-rpath' '--with-uuid=e2fs' '--with-gnu-ld' '--with-pgport=5432' '--with-system-tzdata=/usr/share/zoneinfo' '--with-systemd' 'CFLAGS=-g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer' 'LDFLAGS=-Wl,-z,relro -Wl,-z,now' '--with-gssapi' '--with-ldap' '--with-includes=/usr/include/mit-krb5' '--with-libs=/usr/lib/mit-krb5' '--with-libs=/usr/lib/x86_64-linux-gnu/mit-krb5' '--with-selinux' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2'
```

**Comment by gtod** [comment]
Just FYI, `libbsd` is installed:
```
castor % nm -D --defined-only /lib/x86_64-linux-gnu/libbsd.so.0 |grep arc4
0000000000003fa0 T arc4random
0000000000003d60 T arc4random_addrandom
00000000000041d0 T arc4random_buf
0000000000003d30 T arc4random_stir
0000000000004460 T arc4random_uniform
```

**Comment by tanglebones** [comment]
I will try and take a look this weekend. I think pg_crypto also uses
arc4random, so there must be a configuration trick I am missing.

Also, you can use the pure SQL version, it is fairly fast. (We actually use
that version and not the C one since RDS and Aiven don't let you run custom
plugins.)

**Comment by gtod** [comment]
Indeed, we will use the SQL version (or our own variant) so no need to spend any time on our account.
Thanks again.

**Comment by tanglebones** [comment]
I've created https://github.com/tanglebones/pg_tuid/tree/i4.deb_support that switches `arc4random` for `pg_backend_random` (which I think may only exist in postgresql version 10) since that seems to be the more correct way to get randomness. (I also cleaned up the C90 warnings.) Please let me know if that works for you.

Oddly ossp-uuid is using arc4random still, but it's the only usage I see in the pg sources.

**Comment by gtod** [comment]
This resolves both the warnings I was seeing and the arc4random problem.  So all good now, thank you:
```sql
cusoon=# select tuid_generate();
            tuid_generate             
--------------------------------------
 00056613-9b9b-430e-8800-01c02e5f3733
(1 row)
```

I did forget the `create extension tuid;` line in psql for about 2 minutes, maybe it's worth mentioning in the readme...

**Comment by tanglebones** [comment]
great, merged to master in #050abf4 and I've also updated the readme.
